src/edu/washington/escience/myria/operator/DupElimHashOnly.java

```diff
@@ -0,0 +1,132 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.Arrays;
+import java.util.BitSet;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.primitives.Longs;
+import com.gs.collections.api.iterator.LongIterator;
+import com.gs.collections.impl.list.mutable.primitive.LongArrayList;
+import com.gs.collections.impl.map.mutable.primitive.LongLongHashMap;
+import com.gs.collections.impl.map.mutable.primitive.LongObjectHashMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.util.HashUtils;
+
+/**
+ * A simple implementation of duplicate eliminate. It keeps the references to all the TupleBatches which contain unique
+ * tuples.
+ * */
+public final class DupElimHashOnly extends UnaryOperator {
```

Comment:
I changed the name to `HashDupElim` for now (like "hash aggregator", "hash join", etc.).